### DISCUSSION TITLE
gRPC Web compilation with Emscripten

### DISCUSSION THREAD

**Discussion by moondayyy** [emscripten-core/emscripten#20525] [Q&A]
**Can I compile gRPC to the Web with Emscripten?**
NOTE: I'm trying to compile it but it keeps giving errors, please help,
Thanks.

**Comment by sbc100** [comment]
Please include the specific errors you are seeing when reporting issues like this.

Since gPRC depends on TPC/UDP sockets its unlikely to work well on the web. We do have some solutions in emscripten to getting TCP/UDP code work but they require some extra work and might not be optimal.  See https://emscripten.org/docs/porting/networking.html for more details.
